lms/djangoapps/support/views/manage_user.py

```diff
@@ -66,11 +67,18 @@ def post(self, request, username_or_email):
         user = get_user_model().objects.get(
             Q(username=username_or_email) | Q(email=username_or_email)
         )
+        comment = request.data.get("comment")
+        disable_history = UserAccountDisableHistory.objects.create(
+            user=user, comment=comment, by=request.user
+        )
         if user.has_usable_password():
             user.set_unusable_password()
+            disable_history.disabled = True
         else:
             user.set_password(generate_password(length=25))
+            disable_history.disabled = False
         user.save()
+        disable_history.save()
```

Comment:
Rather than creating `UserAccountDisableHistory` on L#71, initiate variables and create an object at L#81 to avoid to db calls.

Reply:
will do